modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorHighlightSubFetchPhase.java

```diff
@@ -86,7 +87,7 @@ public void process(HitContext hit) throws IOException {
                                 percolatorLeafReaderContext,
                                 slot
                             );
-                            subContext.sourceLookup().setSource(document);
+                            subContext.sourceLookup().setSourceProvider(new SourceLookup.BytesSourceProvider(document));
```

Comment:
One of these days we will get rid of these setters.  Ah well, progress not perfection!